src/cmd/ksh93/edit/edit.c

```diff
@@ -488,8 +487,10 @@ void ed_setup(Edit_t *ep, int fd, int reedit) {
                     int skip = 0;
                     ep->e_crlf = 0;
                     if (pp < ppmax) *pp++ = c;
-                    for (n = 1; *last; n++) {
-                        c = *last++;
+                    for (int n = 1; ; n++) {
+                        c = *last;
+                        if (!c) break;
```

Comment:
`last` was incremented even when `c` is `null`. Are you sure it won't break any other use cases ?

Reply:
Yes, I'm certain this is okay. Specifically because of my comment on your PR. If we reach the nul char and increment `last` the `if (c ==0 || ...) last--;` after this loop undoes the increment. As I've said before, the structure of this is really weird. It's a little less weird with my change as it now looks more like what most people would write.

Note that when I woke up this morning I realized there is a case my change doesn't handle correctly. That is the explicit `continue` statement which would bypass the `last++`.  I'll push a fix for that in a second.